static/js/click_handlers.js

```diff
@@ -547,23 +548,46 @@ export function initialize() {
             onHidden: (instance) => {
                 instance.destroy();
             },
+            onShow: (instance) => {
+                // For both buddy list and top left corner pm list, `target_node`
+                // is their parent `ul` element. We canot use MutationObserver
+                // directly on the reference elmenet because it will be removed
+                // and we need to attach it on an elment which will remain in the
+                // DOM which is their parent `ul`.
+                const target_node = instance.reference.parentElement.parentElement;
+                // We only need to know if any of the `li` elements were removed.
+                const config = {attributes: false, childList: true, subtree: false};
+                const callback = function (mutationsList) {
+                    for (const mutation of mutationsList) {
+                        // Hide instance if reference is in the removed
+                        // node list.
+                        if (
+                            [...mutation.removedNodes].some(
+                                (node) => $(node).has(instance.reference).length !== 0,
+                            )
```

Comment:
I think you need `Array.from(nodes)`, not `[...nodes]`, because Babel in `loose` mode compiles `[...nodes]` to `[].concat(nodes)` which doesn’t do what you want given a `NodeList` rather than an `Array`.

How about `Array.from(mutation.removedNodes).includes(instance.reference.parentElement)`?

Reply:
Thanks, that is great!